packages/dds/tree/src/test/shared-tree/summaryBenchmarkSizes.ts

```diff
@@ -0,0 +1,173 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "assert";
+import { IsoBuffer } from "@fluidframework/common-utils";
+import { makeRandom } from "@fluid-internal/stochastic-test-utils";
+import { TransactionResult } from "../../checkout";
+import { FieldKinds, singleTextCursor } from "../../feature-libraries";
+import { ISharedTree } from "../../shared-tree";
+import { FieldKey, rootFieldKey, rootFieldKeySymbol, TreeValue } from "../../tree";
+import { brand } from "../../util";
+import { TestTreeProvider } from "../utils";
+import { fieldSchema, GlobalFieldKey, namedTreeSchema, SchemaData } from "../../schema-stored";
+// eslint-disable-next-line import/no-internal-modules
+import { PlacePath } from "../../feature-libraries/sequence-change-family";
+
+const globalFieldKey: GlobalFieldKey = brand("globalFieldKey");
+
+describe("Summary size benchmark", () => {
+    it("with no nodes.", async () => {
+        const provider = await TestTreeProvider.create(1);
+        const tree = provider.trees[0];
+
+        const { summary } = tree.getAttachSummary();
+        const summaryString = JSON.stringify(summary);
+        const summarySize = IsoBuffer.from(summaryString).byteLength;
+        assert(summarySize < 1000);
+    });
+    it("with 1 inserted node.", async () => {
+        const summarySize = await getInsertsSummarySize(1, 0, false);
+        assert(summarySize < 2000);
+    });
+    it("with 10 inserted nodes width-wise.", async () => {
+        const summarySize = await getInsertsSummarySize(10, 0, false);
+        assert(summarySize < 3000);
+    });
+    it("with 100 inserted nodes width-wise.", async () => {
+        const summarySize = await getInsertsSummarySize(100, 0, false);
+        assert(summarySize < 20000);
+    });
+    it("with 10 inserted nodes depth-wise.", async () => {
+        const summarySize = await getInsertsSummarySize(10, 0, true);
+        assert(summarySize < 3000);
+    });
+    it("with 100 inserted nodes depth-wise.", async () => {
+        const summarySize = await getInsertsSummarySize(100, 0, true);
+        assert(summarySize < 20000);
```

Comment:
Note of these tests will fail if the summary is empty or your tools to measure it incorrectly report the size (ex: all sizes reported as 0).

it would be a good sanity check to assert something to catch such issues:
```suggestion
        assert(summarySize < 20000);
        assert(summarySize > 1000); // Ensure getInsertsSummarySize actually includes some data from the inserts.
```

Reply:
Added additional assert statement to check that size !== 0

Reply:
Note that assert I suggested was guaranteed to catch if the summary size was always reported as a constant and not actually including the data we are testing (ex: just including the root blob of the summary, or just the schema or something like that).

This is because of constant size was always something less than 1000, this assert would fail, and if it was always some constant 1000 or higher the first test would fail.

As you have it, the tests could all pass if your size measurement doesn't include the actual tree you are testing at all, as long as it includes at least one byte of overhead from something.

Reply:
Thank you Craig!

It turns out that the summaries being returned from the `summarize()` function in `TestTreeProvider` were actually always returning a constant number (as you have mentioned). I have added the `assert(summarySize > 1000);` statement now.

I have also reverted back to using the `getAttachSummary()` function, but am now calling `ensureSynchronized()` as it seems to prevent the `"Clients with local changes cannot be used to generate summaries"` error.